tests/Doctrine/Tests/ORM/Mapping/AttributeReaderTest.php

```diff
@@ -101,3 +122,21 @@ class TestTag
     #[ORM\GeneratedValue]
     public $id;
 }
+
+
+#[Entity]
+#[InheritanceType('SINGLE_TABLE')]
+#[DiscriminatorColumn(name: 'discr', options: ['charset' => 'ascii', 'collation' => 'ascii_general_ci'])]
+#[DiscriminatorMap(['person' => TestPerson::class, 'employee' => TestEmployee::class])]
+class TestPerson
+{
+    #[Id]
+    #[Column(type: 'integer')]
+    #[GeneratedValue]
+    public int|null $id = null;
```

Comment:
While PHP 7.3 and below understands attributes as just comments, it cannot understand this union type.